srcDLL/op2ext.cpp

```diff
@@ -126,13 +126,19 @@ OP2EXT_API size_t GetLoadedModuleName(size_t moduleIndex, char* buffer, size_t b
 
 	std::string moduleName;
 
-	if (moduleIndex < moduleLoader.Count()) {
-		moduleName = moduleLoader.GetModuleName(moduleIndex);
+	try {
+		if (moduleIndex < moduleLoader.Count()) {
+			moduleName = moduleLoader.GetModuleName(moduleIndex);
+		}
+		else if (moduleIndex < GetLoadedModuleCount()) {
+			moduleName = consoleModLoader.GetModuleName();
+		}
+		else {
+			moduleName = "";
+		}
 	}
-	else if (moduleIndex < GetLoadedModuleCount()) {
-		moduleName = consoleModLoader.GetModuleName();
-	}
-	else {
+	catch (...) // Prevent throwing an error across DLL boundaries
+	{
 		moduleName = "";
```

Comment:
The variable `moduleName` is effectively initialized to `""` 3 times in this method. We can remove the later two:
```
std::string moduleName; // Implicitly initialized to ""
moduleName = ""; // Else block (redundant)
moduleName = ""; // Catch block (redundant)
```

The code in the try block has strong exception safety (commit or rollback semantics). That is, if an exception is raised, the code is guaranteed to have no side effects. The only possible side effect is assigning to `moduleName`, but this is done after all possible exceptions are raised. Hence if an exception is raised, the value of `moduleName` should still be it's initial value of `""`.

Similarly, if the `else` block executes, then we've skipped over the other assignments of `moduleName`.

----

We should probably add some logging in the `catch` block. It would be better if errors like this were not silently ignored.

Reply:
I figured you wouldn't like the changes to this function after writing it without logging.

Sometimes I think of string being an object in C# terms which means it isn't auto-initialized. Good point on the redundant else statement. I'll pull it out and add some sort of logging.

Reply:
To be fair, I did like the change, I just noticed we could improve it even further. =)

Interesting to know you already knew what my suggestion would be. From my end it feels like the things I'm able to comment on are forever decreasing in impact.

Reply:
The real question you should be asking is why I didn't just fix it before pushing if I knew it would bother you... :/

Ok, it is coming in the next push.

Reply:
Because not everything needs to be perfect on the first try. Forward progress is still forward progress!

"Forward! Not Backward! ... and always twirling, twirling ..."
https://www.youtube.com/watch?v=0Udvtf77hb0
:wink: